Week5/MyFavoriteMovies/app/src/main/java/com/raywenderlich/myfavoritemovies/activities/LoginActivity.kt

```diff
@@ -0,0 +1,52 @@
+package com.raywenderlich.myfavoritemovies.activities
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.raywenderlich.myfavoritemovies.R
+import kotlinx.android.synthetic.main.activity_login.*
+
+class LoginActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+
+        val sharedPrefs =
+            getSharedPreferences(MainActivity.LOGGED_IN_STATUS_KEY, Context.MODE_PRIVATE)
+        val isLoggedIn = sharedPrefs.getBoolean(MainActivity.IS_LOGGED_IN, false)
+
+        if (isLoggedIn) startMainActivity()
+
+        loginButton.setOnClickListener {
+            if (validateData()) {
+                sharedPrefs.edit().putBoolean(MainActivity.IS_LOGGED_IN, true).apply()
+            }
+        }
+    }
+
+    private fun validateData(): Boolean {
```

Comment:
This function again does too many things. I'd split things like Toasting into a separate function like `showMessage(message: String)`, and have this just return the boolean if the data is valid. Then within a parent function, I'd return which part of the data isn't valid, and show a different message.